contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -240,6 +215,15 @@ func (h *htlcIncomingContestResolver) AttachResolverKit(r ResolverKit) {
 	h.ResolverKit = r
 }
 
+// applyPreimage sets the preimage for the internal htlc resolution and logs the
+// event.
+func (h *htlcIncomingContestResolver) applyPreimage(preimage lntypes.Preimage) {
```

Comment:
Instead of this function, can the inner `htlcSuccessResolver.applyPreimage` be called?

Reply:
i wanted to keep the log message that properly displays the resolver type via `%T`, but yes it is possible